File: 2023/UICPC DIV1/UICPC Round #05/solutions/palindromicpassword/submissions/accepted/palindromicpassword-zf.cpp
#include <iostream>
#include <cstdlib>

using namespace std;

int main() {
    int n;
    cin >> n;
    while (n--) {
        int x;
        cin >> x;

        int best = 0;
        for (int i = 0; i <= 999; ++i) {
            int num = 0, ii = i;
            for (int j = 0; j < 3; ++j, ii /= 10) num = (num*10)+(ii%10);
            num += i*1000;

            if (abs(best-x) > abs(num-x)) best = num;
        }
        cout << best << endl;
    }
    return 0;
}

File: 2023/UICPC DIV1/UICPC Round #05/solutions/palindromicpassword/submissions/accepted/palindromicpassword-zf.py
#!/usr/bin/python3
for x in range(int(input())):
    num = int(input())
    best = 0
    for chop in range(1000):
        tnum = chop*1000 + (chop%10)*100 + (chop%100 - chop%10) + (chop//100)%10
        if abs(tnum-num) < abs(best-num): best = tnum
    print(best)
